### ISSUE TITLE
Warning when SameAs compares expressions of incompatible types

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit.analyzers#154]
```cs
class Foo { }
class Bar { }

var foo = new Foo();
var bar = new Bar();

// Should warn because there is no way that the same instance can be of type Foo and type Bar
Assert.That(foo, Is.SameAs(bar));
```

I think something like bidirectional IsAssignableFrom semantics are desired but I'm not quite sure.

**Comment by Dreamescaper** [comment]
I'll take a look into this, if no objections

**Comment by mikkelbu** [comment]
You are most welcome (currently, most of my time is used on reviewing and screening issues and questions in many of the NUnit repos, so there is little time actually programming).
